src/core/file_format/pkg.cpp

```diff
@@ -0,0 +1,375 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <fstream>
+#include <zlib-ng.h>
+#include "common/io_file.h"
+#include "pkg.h"
+#include "pkg_type.h"
+
+static void DecompressPFSC(std::span<const char> compressed_data,
+                           std::span<char> decompressed_data) {
+    zng_stream decompressStream;
+    decompressStream.zalloc = Z_NULL;
+    decompressStream.zfree = Z_NULL;
+    decompressStream.opaque = Z_NULL;
+
+    if (zng_inflateInit(&decompressStream) != Z_OK) {
+        // std::cerr << "Error initializing zlib for deflation." << std::endl;
+    }
+
+    decompressStream.avail_in = compressed_data.size();
+    decompressStream.next_in = reinterpret_cast<const Bytef*>(compressed_data.data());
+    decompressStream.avail_out = decompressed_data.size();
+    decompressStream.next_out = reinterpret_cast<Bytef*>(decompressed_data.data());
+
+    if (zng_inflate(&decompressStream, Z_FINISH)) {
+    }
+    if (zng_inflateEnd(&decompressStream) != Z_OK) {
+        // std::cerr << "Error ending zlib inflate" << std::endl;
+    }
+}
+
+u32 GetPFSCOffset(std::span<const u8> pfs_image) {
+    static constexpr u32 PfscMagic = 0x43534650;
+    u32 value;
+    for (u32 i = 0x20000; i < pfs_image.size(); i += 0x10000) {
+        std::memcpy(&value, &pfs_image[i], sizeof(u32));
+        if (value == PfscMagic)
+            return i;
+    }
+    return -1;
+}
+
+std::filesystem::path findDirectory(const std::filesystem::path& rootFolder,
+                                    const std::filesystem::path& targetDirectory) {
+    for (const auto& entry : std::filesystem::recursive_directory_iterator(rootFolder)) {
+        if (std::filesystem::is_directory(entry) &&
+            entry.path().filename() == targetDirectory.filename()) {
+            return entry.path();
+        }
+    }
+    return std::filesystem::path(); // Return an empty path if not found
+}
+
+PKG::PKG() = default;
+
+PKG::~PKG() = default;
+
+bool PKG::Open(const std::string& filepath) {
+    Common::FS::IOFile file(filepath, Common::FS::FileAccessMode::Read);
+    if (!file.IsOpen()) {
+        return false;
+    }
+    pkgSize = file.GetSize();
+
+    PKGHeader pkgheader;
+    file.ReadRaw<u8>(&pkgheader, sizeof(PKGHeader));
+
+    // Find title id it is part of pkg_content_id starting at offset 0x40
+    file.Seek(0x47); // skip first 7 characters of content_id
+    file.ReadRaw<u8>(&pkgTitleID, sizeof(pkgTitleID));
```

Comment:
```suggestion
    file.Read(pkgTitleID);
```
Also rename to match codebase style